### ISSUE TITLE
Model.fit in keras style python API doesn't support both x and y being RDDs

### ISSUE THREAD

**Issue by fanzhuyifan** [intel/analytics-zoo#615]
<https://github.com/intel-analytics/analytics-zoo/blob/35fd8d19754296e6e8a9dcdd689c15e88a2e1dab/pyzoo/zoo/pipeline/api/keras/engine/topology.py#L217-L233>

Here, when calling `Model.fit`, if both `x` and `y` are RDDs, an `TypeError("Unsupported training data type: %s" % type(x))` would be thrown.

Also, it would be nice if the documentation specifies the format of the single RDD to be passed, i.e., how the labels and features should be specified, in the case where `y=None`.
